astropy/table/index.py

```diff
@@ -565,9 +552,14 @@ def sort(self):
             self.copy().sort()
 
     def __repr__(self):
-        if self.original:
-            return repr(self.index)
-        return f'Index slice {self.start, self.stop, self.step} of\n{self.index}'
+        col_names = tuple(col.info.name for col in self.index.columns)
+        if len(col_names) == 1:
+            name_str = f'column={col_names[0]!r}'
+        else:
+            name_str = f'columns={col_names}'
+        slice_str = '' if self.original else f' slice={self.start}:{self.stop}:{self.step}'
+        return (f'<{self.__class__.__name__} {name_str} original={self.original}{slice_str}'
```

Comment:
I'd be tempted to be closer to what `__init__` takes, but this is definitely a huge improvement.

Reply:
Good point and I moved the `columns` into the `Index` class where it actually lives. So this moves closer to what's in `__init__`. In the end these classes are not really ever instantiated by users so it's worth being pragmatic about the repr. 

The question is whether the new repr conveys what is needed by users to make (some) sense of this. The previous version was always completely baffling to me and I have repeatedly spent time over the years trying to drill down into `indices` and being confused.